File: apps/android/feature/exercises/ui/src/main/kotlin/com/scribblefit/feature/exercises/ui/components/SetsSection.kt
package com.scribblefit.feature.exercises.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.res.stringResource
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.scribblefit.core.designsystem.ScribbleFitTheme
import com.scribblefit.core.model.Set
import com.scribblefit.feature.exercises.ui.R

@Composable
fun SetsSection(
    sets: List<Set>,
    weightUnit: String
) {
    if (sets.isEmpty()) return

    Column(
        modifier = Modifier.fillMaxWidth()
    ) {
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .background(
                    color = ScribbleFitTheme.colors.surfaceContainerLowest,
                    shape = RoundedCornerShape(ScribbleFitTheme.shapes.medium)
                )
                .padding(ScribbleFitTheme.spacing.medium),
            verticalArrangement = Arrangement.spacedBy(ScribbleFitTheme.spacing.small)
        ) {
            sets.forEach { set ->
                SetItemRow(set = set, weightUnit = weightUnit)
            }
        }
    }
}

@Composable
private fun SetItemRow(
    set: Set,
    weightUnit: String
) {
    Row(
        modifier = Modifier.fillMaxWidth(),
        horizontalArrangement = Arrangement.SpaceBetween,
        verticalAlignment = Alignment.CenterVertically
    ) {
        Row(
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.spacedBy(ScribbleFitTheme.spacing.small)
        ) {
            Text(
                text = "${set.setNumber}.",
                style = ScribbleFitTheme.typography.bodyMedium,
                color = ScribbleFitTheme.colors.midGray,
                fontWeight = FontWeight.Bold
            )
            
            Text(
                text = if ((set.weight ?: 0f) > 0f) "${set.weight} $weightUnit" else "-",
                style = ScribbleFitTheme.typography.bodyLarge,
                color = ScribbleFitTheme.colors.primary,
                fontWeight = FontWeight.Bold
            )
            
            if (set.reps > 0) {
                Text(
                    text = "x",
                    style = ScribbleFitTheme.typography.bodyMedium,
                    color = ScribbleFitTheme.colors.midGray
                )
                Text(
                    text = "${set.reps}",
                    style = ScribbleFitTheme.typography.bodyLarge,
                    color = ScribbleFitTheme.colors.primary,
                    fontWeight = FontWeight.Bold
                )
            }
        }
        
        if (set.rpe != null) {
            Text(
                text = "RPE ${set.rpe}",
                style = ScribbleFitTheme.typography.bodySmall,
                color = ScribbleFitTheme.colors.midGray
            )
        }
    }
}

File: apps/ios/ScribbleFit/LocalPackages/Features/FeatureExercises/Sources/FeatureExercises/UI/Components/SetsSection.swift
import SwiftUI
import CoreModel
import CoreDesignSystem

struct SetsSection: View {
    let sets: [ExerciseSet]
    let weightUnit: String

    var body: some View {
        if !sets.isEmpty {
            VStack(alignment: .leading, spacing: 16) {
                VStack(spacing: 8) {
                    ForEach(sets) { set in
                        SetItemRow(set: set, weightUnit: weightUnit)
                    }
                }
                .padding(16)
                .background(Color.scribbleSurfaceContainerLow)
                .clipShape(RoundedRectangle(cornerRadius: 12))
            }
        }
    }
}

private struct SetItemRow: View {
    let set: ExerciseSet
    let weightUnit: String

    var body: some View {
        HStack {
            HStack(spacing: 8) {
                Text("\(set.setNumber).")
                    .font(ScribbleFitTheme.typography.bodyMedium)
                    .foregroundStyle(Color.scribbleMidGray)
                    .fontWeight(.bold)

                Text(set.weight ?? 0.0 > 0 ? "\(set.weight!.formatted(.number.precision(.fractionLength(0...1)))) \(weightUnit)" : "-")
                    .font(ScribbleFitTheme.typography.bodyLarge)
                    .foregroundStyle(Color.scribblePrimary)
                    .fontWeight(.bold)

                if set.reps > 0 {
                    Text("x")
                        .font(ScribbleFitTheme.typography.bodyMedium)
                        .foregroundStyle(Color.scribbleMidGray)

                    Text("\(set.reps)")
                        .font(ScribbleFitTheme.typography.bodyLarge)
                        .foregroundStyle(Color.scribblePrimary)
                        .fontWeight(.bold)
                }
            }

            Spacer()

            if let rpe = set.rpe {
                Text("RPE \(rpe.formatted(.number.precision(.fractionLength(0...1))))")
                    .font(ScribbleFitTheme.typography.bodySmall)
                    .foregroundStyle(Color.scribbleMidGray)
            }
        }
    }
}
